packages/react-instantsearch-hooks-dom/src/ui/Highlight.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+type HighlightPartProps = {
+  children: React.ReactNode;
+  baseClassName: string;
+  highlightedTagName: React.ReactType;
+  nonHighlightedTagName: React.ReactType;
+  isHighlighted: boolean;
+};
+
+function HighlightPart({
+  baseClassName,
+  children,
+  highlightedTagName,
+  isHighlighted,
+  nonHighlightedTagName,
+}: HighlightPartProps) {
+  const TagName = isHighlighted ? highlightedTagName : nonHighlightedTagName;
+
+  return (
+    <TagName
+      className={
+        isHighlighted
+          ? `${baseClassName}-highlighted`
+          : `${baseClassName}-nonHighlighted`
+      }
+    >
+      {children}
+    </TagName>
+  );
+}
+
+type HighlightedPart = {
+  isHighlighted: boolean;
+  value: string;
+};
+
+export type HighlightProps = React.HTMLAttributes<HTMLSpanElement> & {
+  baseClassName: string;
+  highlightedTagName?: React.ReactType;
+  nonHighlightedTagName?: React.ReactType;
+  separator?: React.ReactNode;
+  parts: HighlightedPart[];
+  partsMultiple: HighlightedPart[][];
```

Comment:
actually, it could always be multiple, we just need to wrap the existing "parts" in an array, the rest of the code can be the same again, will amend tomorrow

Reply:
51f24950